source/patterns/@aws-solutions-constructs/aws-cloudfront-s3/lib/index.ts

```diff
@@ -137,13 +137,17 @@ export class CloudFrontToS3 extends Construct {
     this.s3BucketInterface = originBucket;
 
     // Define the CloudFront Distribution
+    const cloudFrontDistributionForS3Props: defaults.CreateCloudFrontDistributionForS3Props = {
+      sourceBucket: this.s3BucketInterface,
+      cloudFrontDistributionProps: props.cloudFrontDistributionProps,
+      httpSecurityHeaders: props.insertHttpSecurityHeaders,
+      cloudFrontLoggingBucketProps: props.cloudFrontLoggingBucketProps,
+      responseHeadersPolicyProps: props.responseHeadersPolicyProps
+    };
     const cloudFrontDistributionForS3Response = defaults.createCloudFrontDistributionForS3(
       this,
-      this.s3BucketInterface,
-      props.cloudFrontDistributionProps,
-      props.insertHttpSecurityHeaders,
-      props.cloudFrontLoggingBucketProps,
-      props.responseHeadersPolicyProps
+      'cloudfront-dist-for-s3',
```

Comment:
Passing the ID gives us no benefit if we pass a constant value, we'll still get a name collision if we launch multiple constructs in 1 stack. The idea was to pass the id of this instance of the construct (the id argument to the constructor).